earnings/database/Earning/earningsDataGen.js

```diff
@@ -2,75 +2,67 @@ const faker = require('faker');
 
-const tickers = {};
-let allEarnings = [];
 const allQuarters = ['Q4 2017', 'Q1 2018', 'Q2 2018', 'Q3 2018', 'Q4 2018', 'Q1 2019', 'Q2 2019'];
-
-let createTicker = () => {
-  let ticker = '';
-  let alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
-
-  for (let i = 0; i < 5; i++) {
-    ticker += alphabet[Math.floor(Math.random() * 26)]
-  }
-  if (tickers[ticker]) {
-    createTicker();
-  } else {
-    tickers[ticker] = ticker;
-  }
-  return ticker;
-}
+let alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
 
 let generation = () => {
-  const companyData = {
-    ticker: createTicker(),
-    company: faker.company.companyName(),
-  }
-  let companyEarnings = [];
-  let actualEarning = Math.random() * 7;
+  let companyEarnings;
+  let actualEarning;
   let estimatedEarning;
-  let quarterNumber = 0;
-  for (const quarter of allQuarters) {
-    let range = Math.floor(Math.random() * 100);
-    range *= Math.floor(Math.random() * 2) === 1 ? 0.45 : -0.40;
-    actualEarning *= (1 + range / 100);
-    actualEarning = actualEarning.toFixed(2);
+  let quarterNumber;
+  let range;
+  let estimateRange;
+  ((length, limit) => {
+    alphabet = alphabet.split('');
+    let counter = 0;
+    let result;
+    let recurse = (string) => {
+      if (counter === limit) {
+        return;
+      }
+      if (string.length === length) {
+        counter++;
+        const company = {
+          ticker: string,
+          company: faker.company.companyName().replace(/[,]/, ''),
+        }
+        
+        companyEarnings = [];
+        actualEarning = Math.random() * 7;
+        quarterNumber = 0;
+        for (const quarter of allQuarters) {
+          range = Math.floor(Math.random() * 100);
+          range *= Math.floor(Math.random() * 2) === 1 ? 0.45 : -0.40;
+          actualEarning *= (1 + range / 100);
+          actualEarning = actualEarning.toFixed(2);
 
-    let estimateRange = Math.floor(Math.random() * 100);
-    estimateRange *= Math.floor(Math.random() * 2) === 1 ? 0.10 : -0.10;
-    estimatedEarning = actualEarning * (1 + estimateRange / 100);
-    estimatedEarning = estimatedEarning.toFixed(2);
+          estimateRange = Math.floor(Math.random() * 100);
+          estimateRange *= Math.floor(Math.random() * 2) === 1 ? 0.10 : -0.10;
+          estimatedEarning = actualEarning * (1 + estimateRange / 100);
+          estimatedEarning = estimatedEarning.toFixed(2);
 
-    companyEarnings.push({
-      ticker: companyData.ticker,
-      company: companyData.company,
-      quarter,
-      quarterNumber,
-      actualEarning: Number(actualEarning),
-      estimatedEarning: Number(estimatedEarning),
-    });
-    quarterNumber += 1;
-  }
-  allEarnings.push(companyEarnings);
-  return allEarnings;
+          companyEarnings.push(`${company.ticker},${company.company},${quarter},${quarterNumber},${actualEarning},${estimatedEarning}\n`);
+          quarterNumber += 1;
+        }
+        result = companyEarnings.join('');
+        writeOneMillionTimes(file, result);
```

Comment:
@pryz25 I believe this is creating 10 million, so the function should be renamed accordingly.